src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,53 @@
 package mate.academy;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        if (event == null) {
+            throw new IllegalArgumentException("Event cannot be null");
+        }
+        List<Callable<Void>> tasks = new ArrayList<>();
+        for (EventListener listener : listeners) {
+            tasks.add(() -> {
+                listener.onEvent(event);
+                return null;
+            });
+        }
```

Comment:
this seems very strange, the tasks are filled with nulls? let's just do what you had before, a simple for loop where you call `executor.submit(() -> listener.onEvent(event))` or you can also do this via `CompletableFuture.runAsync(() -> listener.onEvent(event), executor)`